### ISSUE TITLE
Unable to publish windows image with foreign layer

### ISSUE THREAD

**Issue by 5ilvers** [jfrog/jfrog-cli#418]
Hi 

I'm trying to push an windows image with `jfrog rt docker-push` to our company's private Artifactory repo.

I keep getting this after jfrog download the manifest of the pushed image
```
[Error] Could not find layer: sha256__65014b3c312172f10bd6701a063f9b5aaf9a916c2d2cb843d406a6f77ded3f8d in Artifactory
```

I downloaded the manifest manually from the Artifactory server, it seems the above layer is one of the two foreign layers in the image.

I'm pushing to a local repo on our Artifactory, and we have a virtual repo that contains both the local repo and a public repo with foreign layer caching disabled. would this be an issue?

**Comment by chatelain-io** [comment]
I have a docker virtual repository with mcr.microsoft.com as remote repository and can see the layer in the remote cache or while I navigate the virtual repository.

But when I use jfrog rt docker-push i'm getting the same issue

**Comment by chatelain-io** [comment]
I found a solution.

You must push your image foreign layers to artifactory.

First configure your docker daemon with: (C:\ProgramData\docker\daemon.json)

```
{
    "allow-nondistributable-artifacts": ["artifactoryrepo"]
}
```

Restart docker daemon

```
docker image pull mcr.microsoft.com/windows/servercore:ltsc2019
docker image tag mcr.microsoft.com/windows/servercore:ltsc2019 artifactory repo/windows/servercore:ltsc2019
docker image push artifactory repo/windows/servercore:ltsc2019
```
jfrog rt docker-push should work after

**Comment by 5ilvers** [comment]
Oh very interesting! I'll give it a shot
Thanks!

**Comment by barbelity** [comment]
@5ilvers, @chatelain-io 
The root cause for this issue is that docker-clients don't push foreign layers to docker registries, resulting in missing foreign layers in Artifactory.

We have released JFrog CLI version 1.32.1 which fixes this issue.
Now, when running docker builds with the CLI, we are not failing the execution in case of missing foreign layers. Please note, that in such case the foreign layer will not be included in the produced build-info.

If you do wish to include all foreign layers in your build-info, the solution you suggested (adding "allow-nondistributable-artifacts": ["artifactoryrepo"] to docker configurations) is the way to do it.
